Dockerfile

```diff
@@ -16,10 +16,6 @@ RUN pip3 install -r requirements/core.txt
 # status of a local service library changes
 ARG service_lib_dir=NO_SUCH_DIR
 
-# Install a local harmony-service-lib-py if we have one
```

Comment:
This was causing the Docker image build to fail. It turns out that I (and presumably Matt) have an empty `deps` directory. But empty directories are not tracked in Git. I think this is primarily a remnant of before `harmony-service-lib` was open-source (maybe?).

Reply:
I think it lets you build a docker image with  a local copy of that lib.  What part was failing? the fact that `deps` doesn't exist?  I think we should keep that ability to build with a local service dir for development.

Reply:
But if you take it out, then the ARG on L17 should come out too?  But I'm pretty sure I've used that feature (and added to it when I was building with local zarr-python

Reply:
I think what was happening with the failure was:

* The job was cloning the repo. That doesn't contain the `deps` directory.
* The Docker image was then built from the `Dockerfile`.
* It got to the line saying `COPY deps ./deps/`, and failed because it couldn't fine the directory in the build environment.

I think we have two options:

1) Add a step further up the workflow that created the `deps` directory is if doesn't exist.
2) Remove these bits and also the `ARG service_lib_dir=NO_SUCH_DIR` that I missed.

I'm in favour of option 2, but could be swayed to option 1. `pip -e ...` is a useful tool for development/debugging, but this is a very specific implementation (e.g., just for the `harmony-service-lib` package) and I reckon someone who is playing with `pip -e ...` probably is also savvy enough to jam a couple of temporary lines into the `Dockerfile` (that bit seems like the easy bit).

Reply:
Yes, I supposed you can convince me that I'll remember how to add that local dependency in the future.  Because I never updated it for python-zarr, just crammed it in there